01-node-tutorial/answers/11-async-js

```diff
@@ -0,0 +1,44 @@
+const fs = require("fs");
+const path = require("path");
+const filePath = path.join(__dirname, "temporary", "fileB.txt");
+
+
+if (!fs.existsSync(path.dirname(filePath))) {
+  fs.mkdirSync(path.dirname(filePath));
+}
+
+console.log("At start");
+
+
+fs.writeFile(filePath, "This is line 1\n", (err) => {
+  console.log("At point 1");
+  if (err) {
+    return console.log("This error happened:", err);
+  }
+
+
+  fs.writeFile(filePath, "This is line 2\n", { flag: "a" }, (err) => {
+    console.log("At point 2");
+    if (err) {
+      return console.log("This error happened:", err);
+    }
+
+    
+    fs.writeFile(filePath, "This is line 3\n", { flag: "a" }, (err) => {
+      console.log("At point 3");
+      if (err) {
+        return console.log("This error happened:", err);
+      }
+
+  
+      fs.readFile(filePath, "utf8", (err, data) => {
+        if (err) {
+          return console.log("Error reading file:", err);
+        }
+        console.log("Final File Content:\n", data);
+      });
+    });
+  });
+});
```

Comment:
It would be better to break out the fs.readFiles into separate functions as it would be easier to read.